### ISSUE TITLE
Model.evaluate progress bar doesn't work in graph mode

### ISSUE THREAD

**Issue by drasmuss** [keras-team/keras#15793]
**System information**.
- Have I written custom code (as opposed to using a stock example script provided in Keras): yes
- OS Platform and Distribution (e.g., Linux Ubuntu 16.04): Windows 10
- TensorFlow installed from (source or binary): binary
- TensorFlow version (use command below): 2.7.0
- Python version: 3.8
- Bazel version (if compiling from source): N/A
- GPU model and memory: N/A
- Exact command to reproduce: See https://colab.research.google.com/gist/drasmuss/b4d074f13c4bed294a1403f351e4d263

**Describe the problem**.

As requested [here](https://github.com/tensorflow/tensorflow/issues/39456#issuecomment-994126508), I am re-opening this issue from the Tensorflow repository here. History/discussion can be seen in that issue.

**Describe the current behavior**.

Calling Model.evaluate in graph mode (after calling `tf.compat.v1.disable_eager_execution()`) doesn't produce any stdout.

**Describe the expected behavior**.

A progress bar should be displayed along with the loss values, as in eager mode.

**[Contributing](https://github.com/keras-team/keras/blob/master/CONTRIBUTING.md)**.

- Do you want to contribute a PR? (yes/no): no
- If yes, please read [this page](https://github.com/keras-team/keras/blob/master/CONTRIBUTING.md) for instructions
- Briefly describe your candidate solution(if contributing):

**Standalone code to reproduce the issue**.

https://colab.research.google.com/gist/drasmuss/b4d074f13c4bed294a1403f351e4d263

**Comment by LukeWood** [comment]
Notes from triage:

We will see if we can reproduce this in the latest version of TensorFlow, we also need to weigh if this is worth fixing considering all of the tf.Session based code will be removed at some point.  This still works in graph mode when you don't disable eager execution.  This is probably not worth fixing as it only occurs when you run tf.compat.disable_eager_execution.

@rchao just in case there is an easy fix here.

**Comment by rchao** [comment]
Bisecting would be the easiest way to figure out a solution. That said I don't think we're prioritizing issues that only occur when eager execution is disabled, though contribution to fix it is welcomed.

Keeping this open for a few days in case there is interest to contribute. Otherwise we should close this issue in a couple of weeks or so.

**Comment by jvishnuvardhan** [comment]
> We will see if we can reproduce this in the latest version of TensorFlow, we also need to weigh if this is worth fixing considering all of the tf.Session based code will be removed at some point. This still works in graph mode when you don't disable eager execution. This is probably not worth fixing as it only occurs when you run tf.compat.disable_eager_execution.

Based on the above comments, I am closing this issue.
